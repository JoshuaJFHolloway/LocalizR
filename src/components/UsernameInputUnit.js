import React from 'react'
import UsernameQuestion from './UsernameQuestion'
import UsernameInput from './UsernameInput'

const UsernameInputUnit = props => {
  return (
    <div>
      <UsernameQuestion/>
      <UsernameInput
        handleUsernameInput = {props.handleUsernameInput}
      />
    </div>
  );
};

export default UsernameInputUnit
