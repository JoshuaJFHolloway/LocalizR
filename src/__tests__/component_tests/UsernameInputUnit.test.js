import React from 'react';
import { shallow } from 'enzyme';
import UsernameInputUnit from '../../components/UsernameInputUnit';

describe('UsernameInputUnit', () => {
  let nameinputunit = shallow(<UsernameInputUnit />);

  describe('Rendering components', () => {

    it('renders correctly', () => {
      expect(nameinputunit ).toMatchSnapshot();
    });

    it('renders NameQuestion component', () => {
      expect(nameinputunit.find('UsernameQuestion').exists()).toBe(true);
    });

    it('renders NameInput component', () => {
      expect(nameinputunit.find('UsernameInput').exists()).toBe(true);
    });

    it('renders NameSubmit component', () => {
      expect(nameinputunit.find('UsernameSubmit').exists()).toBe(true);
    });

  })
});