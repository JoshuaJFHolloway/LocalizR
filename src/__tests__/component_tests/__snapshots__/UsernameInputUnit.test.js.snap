// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernameInputUnit Rendering components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsernameInputUnit />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Question
          dob={undefined}
/>,
        <Input
          handleTextInput={undefined}
          input={undefined}
          value={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dob": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleTextInput": undefined,
          "input": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Question
            dob={undefined}
/>,
          <Input
            handleTextInput={undefined}
            input={undefined}
            value={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "dob": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleTextInput": undefined,
            "input": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
